### ISSUE TITLE
UDP/TCP

### ISSUE THREAD

**Issue by sasmaster** [alm865/FFMPEG-Live555-H264-H265-Streamer#10]
Hi, I am just trying to understand, it streams over UDP or TCP?

**Comment by alm4096** [comment]
Live555 does the streaming. Exactly what it does under the hood i do not
know but usually the RTSP part of the stream is TCP and the underlying
stream is UDP.

In didn't write Live555, please go to their website and check it out.

**Comment by sasmaster** [comment]
Ok.I see,but tell me please, why do you have m_FrameID and m_FrameID2 ,and why they are in range of 32760 ? Is it your stuff or copy paste from live555? I didn't find something similar anywhere else.

**Comment by alm4096** [comment]
The code is all there for you to read, nothing is hidden. Look at
Live_AnalyserSource.cpp, it uses the frame id to check if the frame is a
new frame. m_FrameID and m_FrameID2 are used to store the frame IDs of
frames processed by FFMPEG. At any one time FFMPEG is writing to one frame
and the other is being processed by Live555. It loops in the range 32760
because that's what i chose, you can change it to whatever you like but if
you send 0 it will close down the Live555 thread, and if you send the same
ID twice it will ignore the second. It has to loop at some point, int's
have a finite limit.

Both FFMPEG and LIVE555 are separate threads and you need to manage access
to memory between the two otherwise there will be issues. That's why the
frameIDs exist. There are much better ways to do this but i was still in
the early stages of learning to code at the time.

Everything still works, it just could have been coded far more elegantly.

**Comment by sasmaster** [comment]
Yeah,right,Thread safe Ring buffer queue could replace that logic. I was mostly wondering regarding 32760 number. But now I got it. You just made enough room for unique ids,which are reused in circular manner. Thanks!
